### ISSUE TITLE
Document versioning of C API

### ISSUE THREAD

**Issue by pitrou** [numpy/numpy#6025]
Looking at http://docs.scipy.org/doc/numpy/reference/c-api.html it doesn't seem to be documented which macros and/or magic values to check for the availability of certain APIs (say, for example, I want to access the datetime C APIs, but also provide some compatibility with Numpy 1.6). It would be nice to document that somewhere.

**Comment by pitrou** [comment]
To be clear, I am thinking about how one can make use of NPY_API_VERSION and friends.

**Comment by rgommers** [comment]
Indeed. The only mention anywhere in the reference guide is in http://docs.scipy.org/doc/numpy/reference/c-api.deprecations.html.

**Comment by eayoungs** [comment]
I'm interested in taking on this issue. I'd like to get more involved with a mature open source project and I've heard documentation is a good place to start.
Recently I've been working on implementing some numpy routines in pure python for an open source project related to building simulation, called [eppy](https://github.com/santoshphilip/eppy).

Let me know if you think this would be a good fit.

Thanks

**Comment by charris** [comment]
@eayoungs Can't tell if you are a good fit or not until you give it a shot, there are no formal entrance requirements. You will need to know some C to see how the mechanism works in the include files in `numpy/core/include/numpy`.

**Comment by njsmith** [comment]
As far as the project idea goes, though, yes, it definitely would be
welcome to have some central documentation of what's in which version of
the C API, and the biggest requirement is just the willingness to do some
archaeological investigation through the commit logs, so if you're
interested then I'd say go for it :-).
Another similar situation is for deprecation schedules -- it'd be nice to
have a central list somewhere of each ongoing deprecation, saying when it
started, what changed, what the future plans are, etc. Right now this
information is out there, but scattered a bit haphazardly through the bug
tracker and the source code.

On Mon, Jun 29, 2015 at 3:04 PM, Eric Youngson <notifications@github.com>
wrote:

> I'm interested in taking on this issue. I'd like to get more involved with
> a mature open source project and I've heard documentation is a good place
> to start.
> Recently I've been working on implementing some numpy routines in pure
> python for an open source project related to building simulation, called
> eppy <https://github.com/santoshphilip/eppy>.
>
> Let me know if you think this would be a good fit.
>
> Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/issues/6025#issuecomment-116860397>.
>



-- 
Nathaniel J. Smith -- http://vorpus.org

**Comment by eayoungs** [comment]
Great, I hope to start on it this week.

Cheers!

**Comment by eayoungs** [comment]
Just to keep in touch and for my own tracking purposes, I'm starting work on this today.

**Comment by eayoungs** [comment]
pitrou: Can you point me toward a "good" description of *macros* and/or *magic values* in Python? I'm only familiar with Macros in Excel (VBA) and I've never heard of 'magic values'.

Thanks

**Comment by eayoungs** [comment]
Looks like those are C macros. I'll keep researching...
